### DISCUSSION TITLE
Wagmi Connector and staying connected

### DISCUSSION THREAD

**Discussion by 0xM4m4n** [Web3Auth/Web3Auth#1268] [Help]
Hello!

I seem to be having some issues getting the wallet connected via the web3auth and wagmi connector to stay connected throughout the app session. I am able to successfully connect and sign a message on the sign in page, but anywhere else in the app `useAccount()` from wagmi hook always has undefined as address and `isConnected` false. Is there something I should be looking out for on why this is the case? The wagmi context provider is higher order than any of the internal app components so it should be accessible everywhere.

**Comment by devdomsos** [comment]
I was trying out the native way (without using wagmi connector) of implementing web3Auth and there I saw there was no easy way of stying connected than building your own eagerConnect function that checks in a useEffect if user has logged in already from the storage. Session prop on initialisation of the web3auth instance didn't really help. Could these 2 things be connected?
